src/main/java/com/example/event_manager/controller/BillingController.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@RequestMapping("billings")
+@Controller
+@AllArgsConstructor
+public class BillingController {
+
+  private final EventService eventService;
+  private final BillingRaportService billingRaportService;
+  private final PersonService personService;
+  private final BillingService billingService;
+
+  @GetMapping("/")
+  public void str() {
+    BillingForm billing = BillingForm.builder().title("Title 1").deleted(false)
+        .dateOfEdition(LocalDateTime.now()).dateOfCreation(LocalDateTime.now())
+        .money(BigDecimal.valueOf(200)).personAssigned(personService.findById(1L)).billingType(
+            BillingType.INCOMING).confirmed(true).build();
+    billingService.save(billing);
+  }
+
+  @GetMapping("/details/{id}")
+  public String billingDetails(Model model, @PathVariable Long id) {
+
+    EventForm ef = eventService.eventFormById(id);
+    AccountForm accountForm = new AccountForm();
+    List<BillingForm> billingForms = new ArrayList<>(ef.getBillings());
+    Collections.sort(billingForms);
+
+    BillingFilter billingFilter = new BillingFilter();
+    billingFilter.setDateCreationStart(billingForms.get(0).getDateOfCreation());
+    billingFilter.setDateCreationEnd(billingForms.get(billingForms.size() - 1).getDateOfCreation());
+    accountForm.setBillingFormList(ef.getBillings());
+    accountForm.setEventId(id);
+    model.addAttribute("accountForm", accountForm);
+    model.addAttribute("persons", personService.findAll());
+    model.addAttribute("billingFilter", billingFilter);
+
+    return "billingsDetails";
+  }
+
+  @GetMapping("/details/delete")
+  public String deleteBillingFromEvent(@RequestParam("eventId") Long eventId,
+      @RequestParam("billingId") Long billingId) {
+
+    return "details/delete";
+  }
+
+  @GetMapping("/details/raport")
+  public ResponseEntity<byte[]> billings(@RequestParam final Long id)
+      throws IOException {
+    HttpHeaders headers = new HttpHeaders();
+    EventForm ef = eventService.eventFormById(id);
+    headers.add("Content-Disposition", "attachment; filename=billings.xlsx");
+    final byte[] excelInByteArray = billingRaportService.generateSheetForBillings(ef.getBillings());
+    return ResponseEntity.ok().headers(headers).body(excelInByteArray);
+  }
+
+  @PostMapping("/details/raport/filter")
+  public ResponseEntity<byte[]> getFilterObject(Model model,
+      @ModelAttribute("billingFilter") BillingFilter billingFilter)
+      throws IOException {
+    System.out.println(billingFilter.getPersonName());
+
+    EventForm ef = eventService.eventFormById(1L);
+    List<BillingForm> filteredBillings = billingRaportService
+        .filterBillingFormList(ef.getBillings(), billingFilter);
+    HttpHeaders headers = new HttpHeaders();
+    headers.add("Content-Disposition", "attachment; filename=billings.xlsx");
```

Comment:
use org.springframework.http.HttpHeaders.CONTENT_DISPOSITION

inject billings.xlsx via property value and @value